src/dune_engine/build_system.ml

```diff
@@ -399,10 +399,11 @@ type t =
       -> Build_context.t option
       -> unit Fiber.t
   ; locks : (Path.t, Fiber.Mutex.t) Table.t
-  ; build_mutex : Fiber.Mutex.t option
   ; stats : Stats.t option
   }
 
+let build_mutex = Fiber.Mutex.create ()
```

Comment:
Would be nicer if this was part of `t`. It would make it possible to write unit tests for `Build_system` one day (if we choose to)

Reply:
Make sense, I put it back in `t`